### ISSUE TITLE
An unexpected situation occurred while using AddArc for testing

### ISSUE THREAD

**Issue by JimBobSquarePants** [SixLabors/ImageSharp.Drawing#300]
### Discussed in https://github.com/SixLabors/ImageSharp.Drawing/discussions/298

<div type='discussions-op-text'>

<sup>Originally posted by **inernoro** October 24, 2023</sup>
> I generated some unexpected results when using the following code
Sorry, his center point was accidentally relocated

   ``` [Theory]
    [InlineData(180, 360)]
    [InlineData(180, 359)]
    [InlineData(40, 359)]
    [InlineData(40, 360)]
    [InlineData(10, 359)]
    [InlineData(10, 360)]
    public Image<Rgba32> AddArcTest(int start, int sweep)
    {
        var r = 50;
        var image = new Image<Rgba32>(300, 300, Color.White);
        image.Mutate(x =>
        {
            var centerPoint = new Point(image.Width / 2, image.Height / 2);
            var yuan = new EllipsePolygon(centerPoint.X, centerPoint.Y, r, r);
            x.Draw(Color.Black, 1, yuan);

            var pathBuilder = new PathBuilder();
            var pointF = image.GetCenterPoint();
            pathBuilder.AddArc(pointF, r, r, 0, start, sweep);
            x.Draw(Color.Aqua, 2, pathBuilder.Build());
        });
        image.Save(AsVol($"{start}-{sweep}.png"));
        return image;
    }
```

> 40-360.png

![40-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/4722541a-c3bb-4d42-8368-3b30eb2a4d96)

> 40-359.png

![180-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/4b019644-1e91-424f-ba27-c29953c09b92)

> 180-360.png

![180-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/01326e5e-c529-4970-8a76-12f8030f797b)

> 10-359.png

![10-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/72f82f8d-05ab-4814-8127-892103901c13)

> 10-360 .png

![10-360](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/1b5a00d7-3ca9-4b60-bdd0-04d549cb5bb4)

> 40-359.png

![40-359](https://github.com/SixLabors/ImageSharp.Drawing/assets/9151851/49b295fa-57e6-49af-bffe-9286ee135d91)
</div>

**Comment by JimBobSquarePants** [comment]
The issue is with the naive implementation of circle handling in ArcLineSegment. I have a fix pending.

**Comment by JimBobSquarePants** [comment]
PathBuilder has 89% coverage https://app.codecov.io/gh/SixLabors/ImageSharp.Drawing/blob/main/src%2FImageSharp.Drawing%2FShapes%2FPathBuilder.cs

The tests I’m adding for the fix will increase this. 

We’ll likely create a minor release once I’ve pushed fixes for this and #299

**Comment by JimBobSquarePants** [comment]
v2.0.1 has been released with this fix.
